src/parser/shared/modules/spells/SoulInfusion.tsx

```diff
@@ -0,0 +1,43 @@
+import SPELLS from 'common/SPELLS';
+import CoreChanneling from 'parser/shared/modules/Channeling';
+import Events, {RemoveBuffEvent, CastEvent} from 'parser/core/Events';
+import { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+
+class SoulInfusion extends CoreChanneling {
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER), this.onRemoveBuff);
```

Comment:
```suggestion
    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.SOUL_INFUSION), this.onRemoveBuff);
```

You can do this to filter the events sent to the listener and remove the need to check the spell id first thing in the listener

Reply:
pretty sure i used Hungering/Shriek/etc for the mage ones. and it seems like its removing an entire section from the mage ones?

Reply:
yea i used Shriek for all 3 mage specs. so those snapshots should not be getting changed at all

Reply:
Rewrote my bad code @karlpralow 

I have no idea about the tests tho, reverted them incase I should add or change something @Sharrq

Reply:
So you have to do the tests, cause the PR wont pass without them

The problem is something in your changes is causing other spec's snapshots to change, and it probably shouldnt.

Reply:
Should be better now!